oak-core/src/test/java/org/apache/jackrabbit/oak/plugins/index/AsyncCheckpointServiceTest.java

```diff
@@ -0,0 +1,121 @@
+
+import static org.junit.Assert.*;
+import static org.junit.Assert.assertTrue;
+
+public class AsyncCheckpointServiceTest {
+
+    @Rule
+    public final OsgiContext context = new OsgiContext();
+
+    private final MemoryNodeStore nodeStore = new AsyncCheckpointServiceTest.FakeClusterableMemoryNodeStore();
+    private final AsyncCheckpointService service = new AsyncCheckpointService();
+
+    @Test
+    public void asyncReg() {
+        injectDefaultServices();
+        String name1 = "checkpoint-async-test-1";
+        String name2 = "checkpoint-async-test-2";
+        String name3 = "checkpoint-async-test-3";
+        Map<String,Object> config1 = ImmutableMap.<String, Object>of(
+                "name", "checkpoint-async-test-1",
+                "enable", true,
+                "minConcurrentCheckpoints", 3L,
+                "checkpointLifetime", 60 * 60 * 24L,
+                "timeIntervalBetweenCheckpoints", 60 * 15L
+        );
+
+        Map<String,Object> config2 = ImmutableMap.<String, Object>of(
+                "name", "checkpoint-async-test-2",
+                "enable", false,
+                "minConcurrentCheckpoints", 3L,
+                "checkpointLifetime", 60 * 60 * 24L,
+                "timeIntervalBetweenCheckpoints", 60 * 15L
+        );
+
+        Map<String,Object> config3 = ImmutableMap.<String, Object>of(
+                "name", "checkpoint-async-test-3",
+                "enable", true,
+                "minConcurrentCheckpoints", 4L,
+                "checkpointLifetime", 60 * 24L,
+                "timeIntervalBetweenCheckpoints", 60 * 15L
+        );
+        MockOsgi.activate(service, context.bundleContext(), config1);
+        MockOsgi.activate(service, context.bundleContext(), config2);
+        MockOsgi.activate(service, context.bundleContext(), config3);
+        assertEquals(1, context.getServices(Runnable.class, "(oak.checkpoint.async="+name1+")").length);
+        assertEquals(0, context.getServices(Runnable.class, "(oak.checkpoint.async="+name2+")").length);
+        assertEquals(1, context.getServices(Runnable.class, "(oak.checkpoint.async="+name3+")").length);
+        AsyncCheckpointCreator checkpointCreator1 = getCheckpointCreator("checkpoint-async-test-1");
+        assertEquals(3, checkpointCreator1.getMinConcurrentCheckpoints());
+        assertEquals(60 * 60 * 24, checkpointCreator1.getCheckpointLifetimeInSeconds());
+        AsyncCheckpointCreator checkpointCreator3 = getCheckpointCreator("checkpoint-async-test-3");
+        assertEquals(4, checkpointCreator3.getMinConcurrentCheckpoints());
+        assertEquals(60 * 24, checkpointCreator3.getCheckpointLifetimeInSeconds());
+        MockOsgi.deactivate(service, context.bundleContext());
+        assertNull(context.getService(Runnable.class));
+    }
+
+
+    private AsyncCheckpointCreator getCheckpointCreator(String name) {
+        return (AsyncCheckpointCreator) context.getServices(Runnable.class, "(oak.checkpoint.async="+name+")")[0];
```

Comment:
What is the `[0]` element here?

Reply:
It would be an instance of the AsyncCheckpointCreator ..getServices basically returns a list (even if the match is a single item).